### PR TITLE
feat: flag trends db migration

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Please check inline comments

### CODE REVIEW

File: src/migrations/20240126095544-add-flag-trends.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+exports.up = function(db, cb) {
+    db.runSql(
+        `
+        CREATE TABLE IF NOT EXISTS flag_trends (
+            id VARCHAR(255) NOT NULL,
```
Comment (kwasniew) [inline src/migrations/20240126095544-add-flag-trends.js:7]: this is week id calculated by the app e.g. 2024-03 is the 3rd week of 2024

File: src/migrations/20240126095544-add-flag-trends.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            project VARCHAR(255) NOT NULL REFERENCES projects (id) ON DELETE CASCADE,
+            total_flags INTEGER NOT NULL,
+            stale_flags INTEGER NOT NULL,
+            potentially_stale_flags INTEGER NOT NULL,
+            created_at TIMESTAMP default now(),
+            PRIMARY KEY (id, project)
```
Comment (kwasniew) [inline src/migrations/20240126095544-add-flag-trends.js:13]: week id + project are the unique primary key

File: src/migrations/20240126095544-add-flag-trends.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+exports.up = function(db, cb) {
+    db.runSql(
+        `
+        CREATE TABLE IF NOT EXISTS flag_trends (
+            id VARCHAR(255) NOT NULL,
+            project VARCHAR(255) NOT NULL REFERENCES projects (id) ON DELETE CASCADE,
```
Comment (kwasniew) [inline src/migrations/20240126095544-add-flag-trends.js:8]: I'm thinking whether we should delete trends data when we delete a project. My assumption is that when I delete a project I want to remove its traces in the trends. In the original PR it was no action (https://github.com/Unleash/unleash/pull/5914/files)

File: src/migrations/20240126095544-add-flag-trends.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            total_flags INTEGER NOT NULL,
+            stale_flags INTEGER NOT NULL,
```
Comment (kwasniew) [inline src/migrations/20240126095544-add-flag-trends.js:10]: we separate stale flags marked by user as stale vs potentially stale flags by their age. We can calculate active flags by subtracting stale and potentially stale from total

**Review by FredrikOseberg — APPROVED** [review]

**Pull request merged**